test/test_poolmanager.py

```diff
@@ -247,16 +250,16 @@ def test_http_connection_from_host_case_insensitive(self):
         assert pool is other_pool
         assert all(isinstance(key, PoolKey) for key in p.pools.keys())
 
-    def test_assert_hostname_and_fingerprint_flag(self):
+    def test_assert_hostname_and_fingerprint_flag(self) -> None:
         """Assert that pool manager can accept hostname and fingerprint flags."""
         fingerprint = "92:81:FE:85:F7:0C:26:60:EC:D6:B3:BF:93:CF:F9:71:CC:07:7D:0A"
         p = PoolManager(assert_hostname=True, assert_fingerprint=fingerprint)
         pool = p.connection_from_url("https://example.com/")
         assert 1 == len(p.pools)
-        assert pool.assert_hostname
-        assert fingerprint == pool.assert_fingerprint
+        assert pool.assert_hostname  # type: ignore[attr-defined]
```

Comment:
Why does mypy not know about these attributes?

Reply:
`assert_hostname` and `assert_fingerprint` are attributes of `connectionpool.HTTPSConnectionPool`
https://github.com/urllib3/urllib3/blob/d8eabf304190c62766e05c79ef44be7ff94cfbcb/src/urllib3/connectionpool.py#L873
but we return `HTTPSConnectionPool` from `poolmanager.PoolManager.connection_from_url`
https://github.com/urllib3/urllib3/blob/d8eabf304190c62766e05c79ef44be7ff94cfbcb/src/urllib3/poolmanager.py#L355

We have to change the return type to `HTTPSConnectionPool` for `connection_from_url` and other related functions in `poolmanager.py`

Reply:
Got it

> We have to change the return type to HTTPSConnectionPool for connection_from_url and other related functions in poolmanager.py

I don't think we can do it since it depends on the scheme which is a runtime value.

I think the only thing to improve here is to replace the ignores with an `assert isinstance(pool, HTTPSConnectionPool)`, that would make things clear.